pkg/scheduler/factory.go

```diff
@@ -301,15 +301,16 @@ func (c *Configurator) createFromConfig(policy schedulerapi.Policy) (*Scheduler,
 	defPluginConfig = append(defPluginConfig, pluginConfigForPriorities...)
 	for i := range c.profiles {
 		prof := &c.profiles[i]
-		plugins := &schedulerapi.Plugins{}
-		plugins.Append(&defPlugins)
-		plugins.Apply(prof.Plugins)
-		prof.Plugins = plugins
+		if prof.Plugins != nil {
+			return nil, errors.New("using Plugins and Policy is not supported")
+		}
+		prof.Plugins = &schedulerapi.Plugins{}
+		prof.Plugins.Append(&defPlugins)
 
-		var pluginConfig []schedulerapi.PluginConfig
-		pluginConfig = append(pluginConfig, defPluginConfig...)
-		pluginConfig = append(pluginConfig, prof.PluginConfig...)
-		prof.PluginConfig = pluginConfig
+		if len(prof.PluginConfig) != 0 {
+			return nil, errors.New("using PluginConfig and Policy is not supported")
```

Comment:
Nit: reuse the error string, and `using PluginConfig and Policy at the same time is not supported` sounds clearer.

Reply:
The errors are slightly different. Added the word `simultaneously`.